internal/pkg/api/profile/list.go

```diff
@@ -39,6 +40,22 @@ func ProfileList(ShowOpt *wwapiv1.GetProfileList) (profileList wwapiv1.ProfileLi
 					fmt.Sprintf("%s=%s=%s=%s", p.Id.Print(), f.Field, f.Source, f.Value))
 			}
 		}
+	} else if ShowOpt.ShowJson || ShowOpt.ShowYaml {
+		profileMap := make(map[string]node.NodeInfo)
+		for _, profile := range profiles {
+			profileMap[profile.Id.Get()] = profile
+		}
+
+		buf, _ := yaml.Marshal(profileMap)
+
+		if ShowOpt.ShowJson {
+			jsonBuf, _ := yaml.YAMLToJSON(buf)
+			profileList.Output = append(profileList.Output, string(jsonBuf))
+			return
+		}
+
```

Comment:
Wouldn't this lead to the case that if `json` is selected the `yaml` is always added before the json?